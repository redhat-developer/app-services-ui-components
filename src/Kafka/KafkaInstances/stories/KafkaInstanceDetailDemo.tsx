import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
