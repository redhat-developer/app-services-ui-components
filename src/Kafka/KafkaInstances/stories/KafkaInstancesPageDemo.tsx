import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
