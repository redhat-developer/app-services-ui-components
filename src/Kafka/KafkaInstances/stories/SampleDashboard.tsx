export default () => <div>Dashboard</div>;
