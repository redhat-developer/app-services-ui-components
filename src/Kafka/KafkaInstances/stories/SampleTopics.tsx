export default () => <div>Topics</div>;
