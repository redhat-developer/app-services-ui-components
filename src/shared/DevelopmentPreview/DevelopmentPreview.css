.appserv-devpreview {
  height: 100%;
}
.appserv-devpreview__button {
  --pf-c-button--m-link--Color: var(--pf-global--Color--dark-100) !important;
  text-decoration: underline;
}
