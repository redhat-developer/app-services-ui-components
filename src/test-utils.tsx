import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
