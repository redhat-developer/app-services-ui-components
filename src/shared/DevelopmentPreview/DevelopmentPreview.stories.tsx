import { PageSection } from "@patternfly/react-core";
import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { DevelopmentPreview } from "./DevelopmentPreview";

export default {
  component: DevelopmentPreview,
  args: {
    show: true,
  },
} as ComponentMeta<typeof DevelopmentPreview>;

const Template: ComponentStory<typeof DevelopmentPreview> = (args) => (
  <DevelopmentPreview {...args} />
);

export const ShowDevPreview = Template.bind({});
ShowDevPreview.args = {
  children: (
    <PageSection>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s, when an unknown printer took a galley of type and
        scrambled it to make a type specimen book. It has survived not only five
        centuries, but also the leap into electronic typesetting, remaining
        essentially unchanged. It was popularised in the 1960s with the release
        of Letraset sheets containing Lorem Ipsum passages, and more recently
        with desktop publishing software like Aldus PageMaker including versions
        of Lorem Ipsum.
      </p>
    </PageSection>
  ),
};

export const HideDevPreview = Template.bind({});
HideDevPreview.args = {
  show: false,
  children: (
    <PageSection>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry's standard dummy text ever
        since the 1500s, when an unknown printer took a galley of type and
        scrambled it to make a type specimen book. It has survived not only five
        centuries, but also the leap into electronic typesetting, remaining
        essentially unchanged. It was popularised in the 1960s with the release
        of Letraset sheets containing Lorem Ipsum passages, and more recently
        with desktop publishing software like Aldus PageMaker including versions
        of Lorem Ipsum.
      </p>
    </PageSection>
  ),
};
