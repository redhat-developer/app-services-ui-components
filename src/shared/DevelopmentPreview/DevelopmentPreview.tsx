import type { FunctionComponent } from "react";
import { Banner, Bullseye, Button, Popover } from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import "./DevelopmentPreview.css";

export type DevelopmentPreviewProps = {
  show?: boolean;
};

export const DevelopmentPreview: FunctionComponent<DevelopmentPreviewProps> = ({
  children,
  show,
}) => {
  const { t } = useTranslation();
  if (show) {
    return (
      <div
        id="scrollablePageMain"
        className="pf-c-page__main appserv-devpreview"
        style={{ height: "100%" }}
      >
        <Banner isSticky variant="info">
          <Bullseye>
            <Popover
              aria-label="Service Preview Button"
              hasAutoWidth
              bodyContent={<div>{t("common:servicePreviewTooltip")}</div>}
              position="bottom"
              minWidth="300px"
              maxWidth="25%"
            >
              <Button className="appserv-devpreview__button" variant="link">
                {t("common:servicePreview")}
              </Button>
            </Popover>
          </Bullseye>
        </Banner>
        {children}
      </div>
    );
  }
  return <>{children}</>;
};
