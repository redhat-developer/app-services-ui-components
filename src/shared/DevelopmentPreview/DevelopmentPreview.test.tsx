import { composeStories } from "@storybook/testing-react";
import { render } from "../../test-utils";
import * as stories from "./DevelopmentPreview.stories";

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName!,
  Story,
]);

// Batch snapshot testing
test.each(testCases)("Renders %s story", (_storyName, Story) => {
  const tree = render(<Story />);
  expect(tree.baseElement).toMatchSnapshot();
});
