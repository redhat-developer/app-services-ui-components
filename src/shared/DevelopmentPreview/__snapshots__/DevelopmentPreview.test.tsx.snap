// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HideDevPreview story 1`] = `
<body>
  <div>
    <div
      data-testid="i18n-suspense"
    >
      loading
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders ShowDevPreview story 1`] = `
<body>
  <div>
    <div
      data-testid="i18n-suspense"
    >
      loading
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
